### ISSUE TITLE
About the fitting of marker...

### ISSUE THREAD

**Issue by godrim** [keijiro/BlazeFaceBarracuda#3]
i have download ur project, but the fitting-method of marker is not good. 

the leftEye-to-rightEye-axis can not fitting the change of the marker's rotation and size.  

i think that maybe u should calculate the leftEye-rightEye-nose-plane to fit the marker, but not calculate the leftEye-rightEye-axis.
my algorithm level is not good, i can't  handle the algorithm about the projectin of plane, or other algorithm to optimize the fitting of the maker.

so i submit the issue and hope the algorithm of  maker-fitting will be more perfect.

**Comment by keijiro** [comment]
Sorry but I can't understand what you're trying to suggest. My English language skills are quite limited. Could you rephrase it or add images explaining your concerns?

**Comment by godrim** [comment]
私はあなたのアイテムをダウンロードしましたが、マークの当て方はよくありません。
左目から右目の軸は、マーカーの回転とサイズの変化に適応できません。
マークに合うように左目-右目-鼻の平面を計算するべきかもしれませんが、左目-右目の軸は計算しません。
私のアルゴリズムはレベルが悪いので、平面投影に関するアルゴリズムや、他の最適化メーカーがフィッティングしたアルゴリズムは処理できません。
そこで、私はこの問題を提案し、makerフィッティングアルゴリズムがより改善されることを望んでいます。

**Comment by keijiro** [comment]
It still doesn't make sense to me. Could you show some examples with figures (images)?

**Comment by godrim** [comment]
元の工程では画面が左右逆になっていて、画面を回した後、回転対応がつかなかったので、Markerのマッチング計算に誤りがあったと思います。私は以下の修正をして、やっとマッチングすることができました。しかし、他のアプリケーションシーンでは、イヤリングMarker、帽子Makerなど、マッチング計算は適用されません。
-----------------
1.public class Updater : MonoBehaviour
{

    [HideInInspector] public bool _hflip = false;
}
----------------
2.public sealed class SunglassesUpdater : Updater

-----------------
3.// Rotation
   var angle = Vector2.Angle(diff, Vector2.right);
    if (detection.leftEye.y > detection.rightEye.y) angle *= -1;
     _xform.eulerAngles = Vector3.forward * angle;//先转z轴
     if(_hflip) _xform.eulerAngles = new Vector3(_xform.eulerAngles.x, 180f, _xform.eulerAngles.z);//再转y轴

--------------------
4.// Marker population
   for (var i = 0; i < _markers.Length; i++)
    {
         _markers[i] = Instantiate(_markerPrefab, _previewUI.transform);
         _markers[i].GetComponent<Updater>()._hflip = _hflip;
      }

**Comment by godrim** [comment]
また、画面解像度を1920*1080に変更した後、Markerのマッチング計算は完全に乱れており、Ear、Eye、Nose、Mouthの位置は完全にマッチングしていない。これもマーカーマッチング計算に問題があると思う理由です。

**Comment by keijiro** [comment]
The issues you described don't reproduce on my side.

Could you check if those issues reproduce with the latest version of this repository?

Steps:

1. Open the project.
2. Enable "Sunglasses" game object in the hierarchy.
3. Select the "Image Source" game object and change the "Asset" texture property to "Test7.png".

The attached screenshot below shows the intended behavior.

<img width="789" alt="Screenshot 2023-02-13 at 5 37 42 PM" src="https://user-images.githubusercontent.com/343936/218410362-c94181f3-3488-4539-aff6-99497edddf58.png">

P.S. Please don't use auto translation. This is not a place for personal communication between you and me. The information should be written in the common language for sharing with the community. Sadly, the common language is English, not Japanese.

**Comment by godrim** [comment]
in ur image,
the right-first one, the points of eye/nose/mouth are all moving up, and the maker is moving up. 
the right-second one, the points of eye/ear are not fitting his face, and the maker's rotation is rotating a few degrees of angles clockwise.
when i run ur demo, the situation is more obvious.

**Comment by keijiro** [comment]
> the right-first one, the points of eye/nose/mouth are all moving up, and the maker is moving up.
the right-second one, the points of eye/ear are not fitting his face, and the maker's rotation is rotating a few degrees of angles clockwise.

Those errors are caused by the BlazeFace model. They're not dependent to the sample code.

> when i run ur demo, the situation is more obvious.

Please attach a screenshot to show how obvious it is.

**Comment by keijiro** [comment]
I'm closing this issue due to inactivity. Please reopen it for further problems.
